src/include/sof/audio/module_adapter/library/module_api_ver.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+
+#ifndef __MODULE_API_VER_H__
+#define __MODULE_API_VER_H__
+
+/*
+ *  Api version 5.0.0 for sof loadable modules
+ */
+
+#define SOF_MODULE_API_MAJOR_VERSION 5
+#define SOF_MODULE_API_MIDDLE_VERSION 0
+#define SOF_MODULE_API_MINOR_VERSION 0
+
+union sof_module_api_version {
+	uint32_t full;
+	struct {
+		uint32_t minor    : 10;
+		uint32_t middle   : 10;
+		uint32_t major    : 10;
+		uint32_t reserved : 2;
+	} fields;
+} sof_module_api_version;
+
+struct {
+	uint32_t format;
+	sof_module_api_version api_version_number;
+} sof_module_api_build_info;
```

Comment:
also here looks like you forgot to remove the type name and removed the structure name instead, which made it an object definition instead of a structure declaration